pyHS100/pyHS100.py

```diff
@@ -283,6 +297,168 @@ def identify(self):
 
         return alias, model, features
 
+    @property
+    def alias(self):
+        """
+        Get current device alias (name)
+        :return: Device name aka alias.
+        :rtype: str
+        """
+        return self._alias
+
+    @alias.setter
+    def alias(self, alias):
+        """
+        Sets the device name aka alias.
+        :param alias: New alias (name)
+        :raises SmartPlugException: on error
+        """
+        self._query_helper("system", "set_dev_alias", {"alias": alias})
+
+    @property
+    def led(self):
+        """
+        Returns the state of the led.
+        :return: True if led is on, False otherwise
+        :rtype: bool
+        """
+        return bool(1 - self.sys_info["led_off"])
+
+    @led.setter
+    def led(self, state):
+        """
+        Sets the state of the led (night mode)
+        :param int state: 1 to set led on, 0 to set led off
+        :raises SmartPlugException: on error
+        """
+        self._query_helper("system", "set_led_off", {"off": 1 - state})
+
+    @property
+    def icon(self):
+        """
+        Returns device icon
+        Note: this doesn't seem to work when not using the cloud service, not tested with it either.
+        :return: icon and its hash
+        :rtype: dict
+        :raises SmartPlugException: on error
+        """
+        return self._query_helper("system", "get_dev_icon")
+
+    @icon.setter
+    def icon(self, icon):
+        """
+        Content for hash and icon are unknown.
+        :param str icon: Icon path(?)
+        :raises NotImplementedError: when not implemented
+        :raises SmartPlugError: on error
+        """
+        raise NotImplementedError("Values for this call are unknown at this point.")
+        # here just for the sake of completeness
+        # self._query_helper("system", "set_dev_icon", {"icon": "", "hash": ""})
+
+    @property
+    def time(self):
+        """
+        Returns current time from the device.
+        :return: datetime for device's time
+        :rtype: datetime.datetime
+        :raises SmartPlugException: on error
+        """
+        response = self._query_helper("time", "get_time")
+        return datetime.datetime(response["year"], response["month"], response["mday"], response["hour"], response["min"], response["sec"])
+
+    @time.setter
+    def time(self, ts):
+        """
+        Sets time based on datetime object.
+        Note, this calls set_timezone
+        :param datetime.datetime ts: New date and time
+        :return: result
+        :type: dict
+        :raises NotImplemented: when not implemented.
+        :raises SmartPlugException: on error
+        """
+        raise NotImplementedError("Setting time does not seem to work on HS110 although it returns no error.")
+        """ here just for the sake of completeness / if someone figures out why it doesn't work.
```

Comment:
Um, pep8's 79 chars is more or less a recommendation, but of controversial sort. Can fix this and one below though if wanted.